src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,97 @@
 package core.basesyntax;
 
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        StringBuilder info = new StringBuilder("Report for period ");
+        info.append(dateFrom).append(" - ").append(dateTo);
+
+        for (int i = 0; i < names.length; i++) {
+            int tempSalaryInMonth = 0;
+            for (int j = 0; j < data.length; j++) {
+                String[] tempSpaceSeparatorData = data[j].split(" ");
+                String[] tempPointSeparatorData = tempSpaceSeparatorData[0].split("\\.");
+
+                String[] tempSpaceSeparatorDateFrom = dateFrom.split(" ");
+                String[] tempPointSeparatorDateFrom = tempSpaceSeparatorDateFrom[0].split("\\.");
+
+                String[] tempSpaceSeparatorDateTo = dateTo.split(" ");
+                String[] tempPointSeparatorDateTo = tempSpaceSeparatorDateTo[0].split("\\.");
+
+                int tempYearDateFrom = Integer.parseInt(tempPointSeparatorDateFrom[2]);
+                int tempMonthDateFrom = Integer.parseInt(tempPointSeparatorDateFrom[1]);
+                int tempDayDateFrom = Integer.parseInt(tempPointSeparatorDateFrom[0]);
+
+                int tempYearDateTo = Integer.parseInt(tempPointSeparatorDateTo[2]);
+                int tempMonthDateTo = Integer.parseInt(tempPointSeparatorDateTo[1]);
+                int tempDayDateTo = Integer.parseInt(tempPointSeparatorDateTo[0]);
+
+                int tempYearData = Integer.parseInt(tempPointSeparatorData[2]);
+                int tempMonthData = Integer.parseInt(tempPointSeparatorData[1]);
+                int tempDayData = Integer.parseInt(tempPointSeparatorData[0]);
```

Comment:
Please use DateTimeFormatter and LocalDate with it's .parse() and .isBefore() / .isAfter() / .compareTo() methods to significantly reduce amount of code

Reply:
@olekskov but I have not yet gone through the topic, where there would be "DateTimeFormatter" and these methods. Before the topic "Exceptions" they definitely wasn't

Reply:
found some info and have solved by SimpleDateFormat and Date